cxxopts migration and windows (MinGW) support

edbordin opened the pull request:
As discussed in #38 - the change to cxxopts instead of libargp plus some other minor changes enables us to pretty easily build this for Windows too!

trabucayre commented:
Looks good for me. Just two small thing:
1. Maybe it's better to merge these commit in cxxpts_migration branch
2. a README's update is may be good to detail new options available.

Anyway thank for your contrib!

edbordin commented:
OK, updated the PR to merge to your cxxopts_migrations branch and added a short extra section to the README

Glad I could help, thanks for being willing to support windows!

trabucayre commented:
Applied to cxxopts_migration branch.
Thank for your contribution!
